discovery-provider/src/tasks/entity_manager/entity_manager.py

```diff
@@ -654,3 +677,31 @@ def get_entity_manager_events_tx(update_task, tx_receipt):
     return getattr(
         update_task.entity_manager_contract.events, MANAGE_ENTITY_EVENT_TYPE
     )().processReceipt(tx_receipt)
+
+
+def create_and_raise_indexing_error(err, redis, session):
+    logger.error(
+        f"Error in the indexing task at"
+        f" block={err.blocknumber} and hash={err.txhash}"
+    )
+    # set indexing error
+    set_indexing_error(redis, err.blocknumber, err.blockhash, err.txhash, err.message)
+
+    # seek consensus
+    has_consensus = confirm_indexing_transaction_error(
+        redis, err.blocknumber, err.blockhash, err.txhash, err.message
+    )
+    if not has_consensus:
+        # escalate error and halt indexing until there's consensus
+        error_message = "Indexing halted due to lack of consensus"
+        raise Exception(error_message) from err
+
+    # try to insert into skip tx table
+    skip_tx_hash = save_and_get_skip_tx_hash(session, redis)
+
+    if not skip_tx_hash:
+        error_message = "Reached max transaction skips"
+        raise Exception(error_message) from err
```

Comment:
so if the transaction has been inserted 10 times, retried, then it will halt indexing again?

Reply:
if there's more than 5 transactions skipped, it'll halt indicating something is more severe than a one off error